xwiki-pro-macros-test/xwiki-pro-macros-test-docker/src/test/it/com/xwiki/pro/test/ui/GenericMacrosIT.java

```diff
@@ -76,38 +90,170 @@
+    private void registerMacros()
+    {
         RegisterMacro register = new RegisterMacro();
-        register.registerMacro(BASE_XWIKI_MACRO_SPACE,"Button");
-        register.registerMacro(BASE_XWIKI_MACRO_SPACE,"MicrosoftStream");
-        register.registerMacro(BASE_XWIKI_MACRO_SPACE,"Panel");
+        register.registerMacro(BASE_XWIKI_MACRO_SPACE, "Button");
+        register.registerMacro(BASE_XWIKI_MACRO_SPACE, "MicrosoftStream");
+        register.registerMacro(BASE_XWIKI_MACRO_SPACE, "Panel");
         register.registerMacro(BASE_XWIKI_MACRO_SPACE, "Team");
-        register.registerMacro(BASE_XWIKI_MACRO_SPACE,"Taglist");
-
+        register.registerMacro(BASE_XWIKI_MACRO_SPACE, "Taglist");
     }
 
     @BeforeAll
     void setup(TestUtils setup)
     {
         setup.loginAsSuperAdmin();
-        setup.createUser("UserTest", "UserTest", "");
-        setup.createUser("UserTest2", "UserTest", "");
-        setup.createUser("UserTest3", "UserTest", "");
+        setup.createUser("UserTest", "UserTest", "", "company", "xwiki", "phone", "07777777", "email"
+            , "usertest@example.com", "address", "userTestAddress", "comment", "test", "blog", "https://example.com/",
+            "blogfeed", "https://example.com/");
+        setup.createUser("UserTest2", "UserTest", "", "company", "xwiki", "phone", "07777777", "email"
+            , "usertest2@example.com", "address", "userTestAddress2", "comment", "test2");
+        setup.createUser("UserTest3", "UserTest", "", "company", "xwiki", "phone", "07777777", "email"
+            , "usertest3@example.com", "address", "userTestAddress3", "comment", "test3", "blog", "https://example"
+                + ".com/", "blogfeed", "https://example.com/");
 
+        setup.setGlobalRights("XWiki.XWikiAllGroup", "", "comment", true);
+        setup.setGlobalRights("XWiki.XWikiAllGroup", "", "edit", true);
         setup.deletePage(pageWithTeamMacros);
-        registerMacros();
+        setup.deletePage(pageWithButtonMacros);
+        setup.deletePage(pageWithStatusMacros);
+        setup.deletePage(pageWithTagListMacros);
+        setup.deletePage(pageWithExpandMacro);
+        setup.deletePage(pageWithUserProfileMacro);
+        setup.deletePage(pageWithUserListMacro);
+        setup.deletePage(PageWithPanelMacros);
+        setup.deletePage(pageWithTags);
+        setup.deletePage(PageWithMStreamMacros);
 
+        setup.createPage(pageWithTags, "Test content for tagging");
+        setup.gotoPage(pageWithTags);
+        TaggablePage taggablePage = new TaggablePage();
+        AddTagsPane tagsPane = taggablePage.addTags();
+        tagsPane.setTags("alpha, beta, gamma");
+        tagsPane.add();
+
+        setup.createPage(pageWithTags2, "Test content for tagging");
+        setup.gotoPage(pageWithTags2);
+        TaggablePage taggablePage2 = new TaggablePage();
+        AddTagsPane tagsPane2 = taggablePage2.addTags();
+        tagsPane2.setTags("z, x, y");
+        tagsPane2.add();
```

Comment:
Are these needed for all / multiple tests, or only for `tagListMacroTest`? If it's only for `tagListMacroTest`, I would move it to a separate method and call it only for that test.